lib/spack/docs/packaging_guide.rst

```diff
@@ -617,30 +617,39 @@ it executable, then runs it with some arguments.
 Deprecating old versions
 ^^^^^^^^^^^^^^^^^^^^^^^^
 
-There are many reasons to remove old versions of software:
-
-#. Security vulnerabilities (most serious reason)
-#. Changing build systems that increase package complexity
-#. Changing dependencies/patches/resources/flags that increase package complexity
-#. Maintainer/developer inability/unwillingness to support old versions
-#. No longer available for download (right to be forgotten)
-#. Package or version rename
-
-At the same time, there are many reasons to keep old versions of software:
-
-#. Reproducibility
-#. Requirements for older packages (e.g. some packages still rely on Qt 3)
-
-In general, you should not remove old versions from a ``package.py``. Instead,
-you should first deprecate them using the following syntax:
+There are many reasons to remove old versions of software (security vulnerabilities,
+severe bugs, changing build systems, etc.) and at the same time, there are many reasons
+to keep old versions of software (reproducibility, maintain requirements for older
+packages). In general, you should not remove old versions from a ``package.py`` directly
+but instead, you should first deprecate them using the following syntax:
 
 .. code-block:: python
 
+   version('1.2.4', sha256='...')
    version('1.2.3', sha256='...', deprecated=True)
-
-
-This has two effects. First, ``spack info`` will no longer advertise that
-version. Second, commands like ``spack install`` that fetch the package will
+   version('1.2.2', sha256='...', deprecated='vuln')
+
+In the example above we deprecated version ``1.2.3`` and version ``1.2.2`` of the software.
+The ``deprecated=`` argument of the version directive can be either a boolean (in which case
+``True`` marks the version as deprecated without giving additional information on the deprecation
+reason) or a string representing the reason of deprecation. Conventionally in Spack we use the
+following reasons:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Value of the ``deprecated=`` argument
+     - Reason of deprecation
+   * - 'vuln'
+     - Security vulnerabilities (most serious reason)
+   * - 'bug'
+     - Serious bugs reported
+   * - 'unavailable'
+     - No longer available for download
```

Comment:
Should there be an "unsupported" category indicating that the version series is no longer maintained and may eventually be incompatible with newer OS frameworks (often seen for GUI  packages on macOS) or more strict compilers?

Reply:
I'm absolutely open to modify this list, either adding new reasons or removing ones that are already there.The current implementation allows the use of any string as a deprecation reason, so adding new reasons in the future should just be matter of documenting them.